server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java

```diff
@@ -1011,19 +1026,32 @@ private void setUpdateTablet(UpdateSession us, KeyExtent keyExtent) {
         updateMetrics.addPermissionErrors(0);
         return;
       }
+
     }
 
     @Override
     public void applyUpdates(TInfo tinfo, long updateID, TKeyExtent tkeyExtent,
-        List<TMutation> tmutations) {
+        List<TMutation> tmutations) throws TException {
       UpdateSession us = (UpdateSession) sessionManager.reserveSession(updateID);
       if (us == null) {
         return;
       }
 
       boolean reserved = true;
+      boolean allowWriteThreadSemaphore = false;
```

Comment:
minor nit: call this something like semaphoreAcquired.  Then the check at line 1083 would be if (semaphoreAcquired) which would be much more readable.